src/Features/CSharp/Portable/UseIndexOrRangeOperator/CSharpUseRangeOperatorCodeFixProvider.cs

```diff
@@ -0,0 +1,216 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseIndexOrRangeOperator
+{
+    using static CSharpUseRangeOperatorDiagnosticAnalyzer;
+    using static Helpers;
+    using static SyntaxFactory;
+
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpUseRangeOperatorCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.UseRangeOperatorDiagnosticId);
+ 
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics[0], c)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            foreach (var diagnostic in diagnostics)
+            {
+                FixOne(semanticModel, editor, diagnostic, cancellationToken);
+            }
+        }
+
+        private void FixOne(
+            SemanticModel semanticModel, SyntaxEditor editor,
+            Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var invocation = (InvocationExpressionSyntax)diagnostic.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken);
+            var expression = invocation.Expression is MemberAccessExpressionSyntax memberAccess
+                ? memberAccess.Expression
+                : invocation.Expression;
+
+            var rangeExpression = CreateRangeExpression(
+                semanticModel, diagnostic, invocation, cancellationToken);
+            var argument = Argument(rangeExpression).WithAdditionalAnnotations(Formatter.Annotation);
+            var arguments = SingletonSeparatedList(argument);
+
+            if (diagnostic.Properties.ContainsKey(UseIndexer))
+            {
+                var argList = invocation.ArgumentList;
+                var elementAccess = ElementAccessExpression(
+                    expression,
+                    BracketedArgumentList(
+                        Token(SyntaxKind.OpenBracketToken).WithTriviaFrom(argList.OpenParenToken),
+                        arguments,
+                        Token(SyntaxKind.CloseBracketToken).WithTriviaFrom(argList.CloseParenToken)));
+
+                editor.ReplaceNode(invocation, elementAccess);
```

Comment:
>editor.ReplaceNode(invocation, elementAccess); [](start = 16, length = 46)

Can FixAll throw here for nested invocation case where both the inner and outer invocations have the diagnostic?

Reply:
Yes.  It would be complex to construct, but would likely be possible.  Let me handle that.

Reply:
Seems fine to even just detect and bail out on one of the diagnostics as this will be rare and next FixAll attempt should fix it.

---
In reply to: [227546966](https://github.com/dotnet/roslyn/pull/30635#discussion_r227546966) [](ancestors = 227546966)